### DISCUSSION TITLE
Dynamo DB taking long time to fetch record with high latency up to 5-6 second

### DISCUSSION THREAD

**Discussion by kmohit520** [aws/aws-sdk-java-v2#3298] [Q&A]
**pom.xml**

`<dependency>`
    `  <groupId>com.amazonaws</groupId>`
    `<artifactId>aws-java-sdk-bom</artifactId>`
    `<version>1.11.256</version>`
`</dependency>`
`<dependency>`
    `<groupId>com.amazonaws</groupId>`
    `<artifactId>aws-java-sdk-dynamodb</artifactId>`
`</dependency>`

**Client Code**

`public static AmazonDynamoDB getDynamoDBClient() {`
       `return AmazonDynamoDBClientBuilder.standard().withRegion("ap-southeast-1").build();`
`}
`
Now i am trying to execute a normal query having few records but it is taking long time to fetch the result.

For first time it is fetching the records in around 5-6 seconds on multiple requests in reduces by half. 2-3 seconds is still large time for fetching only few items.

Already checked the tuning of dynamo DB using different client configurations (connection timeout, request timeout, retry etc.) but not giving results as expected.
Also checked with SDK version 2 URLConnectionHTTPClient config but same results came there too.

One possible cause can be the credentials fetch time for dynamo DB client but not having any credentials caching reference in java. Can any one suggest possible configuration to improve this latency.

**Comment by yasminetalby** [comment]
Hello @kmohit520 ,

Thank you for your question.

AWS has a feature that allows you to capture and visualize metrics related to the runtime environment which can help to identify the cause of the latency issue encountered. This feature is disabled by default but you can enable it following the documentation : https://aws.amazon.com/blogs/developer/enabling-metrics-with-the-aws-sdk-for-java/
Once we have more information regarding the origin of the latency, we can look into potential configuration to improve your it.

Best,
Yasmine

**Reply by kmohit520** [reply]
Hi @yasminetalby Thanks for quick response.

Actually I tried to enable the metrics by adding below configuration in application.properties file:
 `-Dcom.amazonaws.sdk.enableDefaultMetrics=credentialFile=/path/aws.properties`

But not sure if it is the right place to add this config. Also we are not using aws.properties in our project.
So could you please refer any example repo for reference.

Best Regards
Mohit

**Comment by yasminetalby** [comment]
Hello @kmohit520 ,

My apology for the delayed answer. 
You can see an example on how to enable metrics in the Gathering Metrics section : https://aws.amazon.com/blogs/developer/tuning-the-aws-sdk-for-java-to-improve-resiliency/

Best Regards,
Yasmine
